hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/rollback/ListingBasedRollbackStrategy.java

```diff
@@ -194,19 +194,15 @@ private String getBaseFileExtension(HoodieTableMetaClient metaClient) {
   }
 
   @NotNull
-  private HoodieRollbackRequest getHoodieRollbackRequest(String partitionPath, FileStatus[] filesToDeletedStatus) {
-    List<String> filesToDelete = getFilesToBeDeleted(filesToDeletedStatus);
-    return new HoodieRollbackRequest(
-        partitionPath, EMPTY_STRING, EMPTY_STRING, filesToDelete, Collections.emptyMap());
-  }
-
-  @NotNull
-  private List<String> getFilesToBeDeleted(FileStatus[] dataFilesToDeletedStatus) {
-    return Arrays.stream(dataFilesToDeletedStatus).map(fileStatus -> {
-      String dataFileToBeDeleted = fileStatus.getPath().toString();
-      // strip scheme E.g: file:/var/folders
-      return dataFileToBeDeleted.substring(dataFileToBeDeleted.indexOf(":") + 1);
-    }).collect(Collectors.toList());
+  private List<HoodieRollbackRequest> getHoodieRollbackRequests(String partitionPath, FileStatus[] filesToDeletedStatus) {
+    return Arrays.stream(filesToDeletedStatus)
+        .map(fileStatus -> {
+          String dataFileToBeDeleted = fileStatus.getPath().toString();
+          // strip scheme E.g: file:/var/folders
```

Comment:
So the parallelism is configured to as file granuality(currently it is partition based rollback), and now the max parallelism could be the max number of files to clean:

```java
Math.max(Math.min(rollbackRequests.size(), config.getRollbackParallelism()), 1)
```

Reply:
yes, thats my understanding as well.

Reply:
does this cause any issues while deserializing HoodieRollbackPlan ? (in other words, we should be backwards compatible with whats in master right?

Reply:
All the avro data model should already be serializable, including `HoodieRollbackPlan`.